packages/teleterm/src/logger.ts

```diff
@@ -1,51 +1,36 @@
 import * as types from 'teleterm/types';
 
-export class DefaultService {
-  createLogger(loggerName: string): types.Logger {
-    const name = loggerName;
-
-    const log = (level = 'log', ...args) => {
-      console[level](`%c[${name}]`, `color: blue;`, ...args);
-    };
-
-    return {
-      warn(...args: any[]) {
-        log('warn', ...args);
-      },
-
-      info(...args: any[]) {
-        log('info', ...args);
-      },
-
-      error(...args: any[]) {
-        log('error', ...args);
-      },
-    };
-  }
-}
-
 export default class Logger {
+  private static service: types.LoggerService;
   private logger: types.Logger;
 
-  private static service = new DefaultService();
-
-  constructor(context = '') {
-    this.logger = Logger.service.createLogger(context);
-  }
+  constructor(private context = '') {}
 
   warn(message: string, ...args: any[]) {
-    this.logger.warn(message, ...args);
+    this.getLogger().warn(message, ...args);
   }
 
   info(message: string, ...args: any[]) {
-    this.logger.info(message, ...args);
+    this.getLogger().info(message, ...args);
   }
 
   error(message: string, ...args: any[]) {
-    this.logger.error(message, ...args);
+    this.getLogger().error(message, ...args);
   }
 
   static init(service: types.LoggerService) {
     Logger.service = service;
   }
+
+  private getLogger(): types.Logger {
+    if (!Logger.service) {
+      throw new Error('Logger is not initialized');
+    }
```

Comment:
We can't guarantee, but it's on purpose.
Previously, the logger switched to `DefaultService` silently, now you can see that something is wrong because of the error.
And we don't have much choice here. We could disallow creating loggers when they are imported, but it's inconvenient for the places like [File Storage](https://github.com/gravitational/webapps/blob/master/packages/teleterm/src/services/fileStorage/fileStorage.ts) (when you create one logger for all the functions). 
IMO the way with lazy-initialing the loggers is a better option here.

Reply:
I agree with you completely.
I'll keep in mind that we have to make the improvements here, but tbh I can't find any elegant solution so far :(